clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -527,4 +525,78 @@ public void onComplete(ClientResponse response) {
         boolean shouldBlock();
     }
 
+
+    /*
+ * A threadsafe helper class to hold RecordBatches that haven't been ack'd yet
+ */
+    private final static class UnsentRequests {
+        private final Map<Node, ConcurrentLinkedQueue<ClientRequest>> unsent;
```

Comment:
Good point. It is not needed. At some point I thought it needed in order to expose iterator of the unset outside class UnsentRequests. But the final implementation turns out that it is not needed. Changed to List now.